File: VB/S171806_Build.Web/ReportBuildInterceptor.vb
Imports Microsoft.VisualBasic
Imports System.ComponentModel.Composition
Imports DevExpress.XtraReports.Service.Extensions
Imports DevExpress.XtraReports.UI

Namespace S171806_Build.Web
	<Export(GetType(IReportBuildInterceptor))> _
	Public Class ReportBuildInterceptor
		Implements IReportBuildInterceptor
		#Region "IReportBuildInterceptor Members"
		Private Sub InvokeBefore(ByVal report As XtraReport, ByVal customArgs As Object) Implements IReportBuildInterceptor.InvokeBefore
			report.Watermark.Text = "Immutable Text"
			report.Watermark.ShowBehind = False
			report.Watermark.TextTransparency = 0
		End Sub

		Private Sub InvokeAfter(ByVal report As XtraReport, ByVal customArgs As Object) Implements IReportBuildInterceptor.InvokeAfter
		End Sub
		#End Region
	End Class
End Namespace

File: CS/S171806_Build.Web/ReportBuildInterceptor.cs
using System.ComponentModel.Composition;
using DevExpress.XtraReports.Service.Extensions;
using DevExpress.XtraReports.UI;

namespace S171806_Build.Web {
    [Export(typeof(IReportBuildInterceptor))]
    public class ReportBuildInterceptor : IReportBuildInterceptor {
        #region IReportBuildInterceptor Members
        void IReportBuildInterceptor.InvokeBefore(XtraReport report, object customArgs) {
            report.Watermark.Text = "Immutable Text";
            report.Watermark.ShowBehind = false;
            report.Watermark.TextTransparency = 0;
        }

        void IReportBuildInterceptor.InvokeAfter(XtraReport report, object customArgs) {
        }
        #endregion
    }
}